### DISCUSSION TITLE
Possibly add `XCTestExpectation` to dependency overrides in `TestStore`?

### DISCUSSION THREAD

**Discussion by Berhtulf** [pointfreeco/swift-composable-architecture#1730] [Ideas]
Hello,
what do you all think about adding implicit `XCTestExpectation` when we manually override dependency endpoint in `TestStore`? I don't have any solutions ready for this, since its just an Idea I had, when testing `Void` returning Clients.

If I am not wrong, now the only way to prove that a dependency was called, we have to add Expectations manually like this

```
let expectation = expectation(description: "Description text")
testStore.dependencies.userDefaultsClient.setString = { _, _ in
  expectation.fulfill()
}

await testStore.send(.onAppear) {
  ...
}

wait(for: [expectation], timeout: 0.1)

```

I am thinking about some naive way where I can just state that this dependency is implemented using current syntax (possibly enhanced with a timeout value), and TCA would force me to call it when exhaustive testing is turned on. Like it does when this endpoint is called, but I did not provide it, it fails the tests, when using `XCTUnimplemented`

**Comment by ibrahimkteish** [comment]
I usually do the following:
```swift 
var didCallSetString = false
testStore.dependencies.userDefaultsClient.setString = { _, _ in
didCallSetString = true
}

await testStore.send(.onAppear) {
  ...
}

XCTAssertTrue(didCallSetString)
```

Without using timeout.

**Comment by kuglee** [comment]
You should look into ActorIsolated. There's an example in the [docs](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/actorisolated). about how to use it.
